pkg/kubectl/bind-apiservice/plugin/servicebindings.go

```diff
@@ -139,13 +141,182 @@ func (b *BindAPIServiceOptions) createAPIServiceBindings(ctx context.Context, co
 	return bindings, nil
 }
 
+//Print Permission Claim to provided Writer
+func printPermissionClaim(w io.Writer, p kubebindv1alpha1.PermissionClaim) error {
+	var b bytes.Buffer
+
+	var groupResource string
+	if p.GroupResource.Group != "" {
+		groupResource = fmt.Sprintf("%s objects (apiVersion: \"%s/%s\")", p.GroupResource.Resource, p.GroupResource.Group, p.Version)
+	} else {
+		groupResource = fmt.Sprintf("%s objects (apiVersion: \"%s\")", p.GroupResource.Resource, p.Version)
+	}
+
+	if err := writeFirstLines(&b, groupResource, p); err != nil {
+		return err
+	}
+
+	if err := writeOnConflict(&b, p); err != nil {
+		return err
+	}
+
+	if err := writeUpdateClause(&b, p); err != nil {
+		return err
+	}
+
+	if err := writeRequiredAndAcceptance(&b, p.Required); err != nil {
+		return err
+	}
+
+	_, err := fmt.Fprint(w, b.String())
+	return err
+}
+
+func writeFirstLines(b *bytes.Buffer, groupResource string, claim kubebindv1alpha1.PermissionClaim) error {
+	var err error
+
+	selector := claim.Selector
+
+	switch {
+	case selector == kubebindv1alpha1.ResourceSelector{} || selector.Owner == "":
+		selector.Owner = kubebindv1alpha1.Consumer
+		err = writeConsumerCase(b, groupResource, claim)
+	case selector.Owner == kubebindv1alpha1.Provider:
+		err = writeProviderCase(b, groupResource, claim)
+	case selector.Owner == kubebindv1alpha1.Consumer:
+		err = writeConsumerCase(b, groupResource, claim)
+	}
+	return err
+}
+
+func writeProviderCase(b *bytes.Buffer, groupResource string, claim kubebindv1alpha1.PermissionClaim) error {
+	var err error
+
+	donate := false
+	if claim.Create != nil {
+		donate = claim.Create.Donate
+	}
+
+	selectorName := claim.Selector.Name
+
+	switch {
```

Comment:
This string building can be split up to reduce duplication.
For example 
```go
fmt.Fprintf(b, "Create and modify %s on your cluster", <"user-owned"/"provider owned">)
fmt.Fprintf(b, "referenced with: %s", ...)
fmt.Fprintf(b, "\n")
```

Reply:
I simplified this in 926bb5993775516f6dbfe28787228a9b11d95f92